app/builders/messages/instagram/message_builder.rb

```diff
@@ -69,9 +69,27 @@ def contact
   end
 
   def conversation
-    @conversation ||= Conversation.where(conversation_params).find_by(
-      "additional_attributes ->> 'type' = 'instagram_direct_message'"
-    ) || build_conversation
+    @conversation ||= set_conversation_based_on_inbox_config
+  end
+
+  def set_conversation_based_on_inbox_config
+    if @inbox.lock_to_single_conversation
+      Conversation.where(conversation_params)
+                  .find_by("additional_attributes ->> 'type' = 'instagram_direct_message'") || build_conversation
```

Comment:
The common query can be abstracted out.

Reply:
@scmmishra refactored the part. Let me know if anything else needs to be changed.